src/main/scala/org/clulab/wm/discovery/ConceptDiscovery.scala

```diff
@@ -46,6 +51,31 @@ class ConceptDiscovery {
     }
   }
 
+  /** Currently pruning sentences from the text whose Qntfy sentence score are below a threshold.
+    * TODO: should this be a top k instead? */
+  def mkDocument(processor: Processor, text: String, sentenceThreshold: Option[Double], rankedSentences: Seq[SentenceScore]): Document = {
+    if (sentenceThreshold.isEmpty) {
+      processor.annotate(text)
+    }
+    else {
+      val document = processor.mkDocument(text)
+      val lastValid = rankedSentences.indexWhere(_.score < sentenceThreshold)
+      // only accept sentences with a saliency above some threshold
+      val validSentences = rankedSentences.slice(0, lastValid)
+      val sentencesByOffsets = document.sentences.map(s => ((s.startOffsets.head, s.endOffsets.last), s)).toMap
+      val keptSentences = validSentences
+        // keep ones that are in the validSentences
+        .map(s => sentencesByOffsets((s.start, s.end)))
+        // put them back in order
+        .sortBy(_.startOffsets.head)
+        // get their text
+        .map(_.getSentenceText)
+        .toArray
```

Comment:
If `keptSentences` is just an `Iterable[String]`, why not just:

    val keptSentences = rankedSentences.filter(_.score <= sentenceThreshold).sortBy(_.start).map {
        case SentenceScore(start, end, _) => text.slice(start, end)
    }